commit 3950cbd7a18b48c0f272ca41fa60cc135ff3175a
Author: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date:   Wed Mar 20 11:31:54 2024 -0300

    math: Fix i386 and m68k exp10 on static build (BZ 31775)
    
    The commit 08ddd26814 removed the static exp10 on i386 and m68k with an
    empty w_exp10.c (required for the ABIs that uses the newly
    implementation).  This patch fixes by adding the required symbols on the
    arch-specific w_exp{f}_compat.c implementation.
    
    Checked on i686-linux-gnu and with a build for m68k-linux-gnu.
    
    Reviewed-by: Aurelien Jarno <aurelien@aurel32.net>
    (cherry picked from commit 1f09aae36aa185b8b1100dfa6b776442279bf580)

diff --git a/sysdeps/i386/fpu/w_exp10_compat.c b/sysdeps/i386/fpu/w_exp10_compat.c
index b53455386ed8e189..49a0e03385e4bfe3 100644
--- a/sysdeps/i386/fpu/w_exp10_compat.c
+++ b/sysdeps/i386/fpu/w_exp10_compat.c
@@ -1,3 +1,8 @@
 /* i386 provides an optimized __ieee754_exp10.  */
-#define NO_COMPAT_NEEDED 1
-#include <math/w_exp10_compat.c>
+#ifdef SHARED
+# define NO_COMPAT_NEEDED 1
+# include <math/w_exp10_compat.c>
+#else
+# include <math-type-macros-double.h>
+# include <w_exp10_template.c>
+#endif
diff --git a/sysdeps/m68k/m680x0/fpu/w_exp10_compat.c b/sysdeps/m68k/m680x0/fpu/w_exp10_compat.c
index 0d3e7186261041e3..350f2e4b4d37e569 100644
--- a/sysdeps/m68k/m680x0/fpu/w_exp10_compat.c
+++ b/sysdeps/m68k/m680x0/fpu/w_exp10_compat.c
@@ -1,3 +1,8 @@
 /* m68k provides an optimized __ieee754_exp10.  */
-#define NO_COMPAT_NEEDED 1
-#include <math/w_exp10_compat.c>
+#ifdef SHARED
+# define NO_COMPAT_NEEDED 1
+# include <math/w_exp10_compat.c>
+#else
+# include <math-type-macros-double.h>
+# include <w_exp10_template.c>
+#endif
